playground/csp/__tests__/csp.spec.ts

```diff
@@ -27,6 +27,20 @@ test('dynamic js', async () => {
   )
 })
 
+test('inline js', async () => {
+  await expectWithRetry(() => page.textContent('.inline-js')).toBe(
+    'inline-js: ok',
+  )
+})
+
+test('nonce attributes are not repeated', async () => {
+  const htmlSource = await page.content()
+  expect(htmlSource).not.toContain(/nonce=""[^>]*nonce=""/)
```

Comment:
`#$NONCE$#` is replaced with a non-empty string. So `nonce=""` won't appear and this `expect` won't fail even without the fix.
I guess it needs to be something like:
```js
expect(htmlSource).not.toContain(/nonce="[^"]+"[^>]*nonce="[^"]+"/)
```

Reply:
I don't believe that is true. The browser strips the attribute value for security reasons and the only way to see it is to access `.nonce` from the javascript side. I console logged the result of `page.content()` when writing this test and confirmed that it comes back as `nonce=""`

I could change your regex to use `*` instead of `+` in the attribute value and then this would work if the browser in test ever started returning content. I'll leave that up to you to decide.

Reply:
I'm using `page.content()` as a proxy for the source that was sent to the browser from the server which _would_ contain the filled out nonce value. I'm not sure if there is a better way to grab that.

Reply:
Ah, your right. I forgot about the nonce hiding behavior. 🤦 Thanks for clarifying.